source/common/router/router.cc

```diff
@@ -303,6 +316,14 @@ Http::FilterTrailersStatus Filter::decodeTrailers(Http::HeaderMap& trailers) {
   return Http::FilterTrailersStatus::StopIteration;
 }
 
+void Filter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& callbacks) {
+  callbacks_ = &callbacks;
+  // As the decoder filter only pushes back via watermarks once data has reached
+  // it, it can latch the current buffer limit and does not need to update the
+  // limit if another filter increases it.
+  buffer_limit_ = callbacks_->decoderBufferLimit();
```

Comment:
random thought: Should we assert inside connection manager that set/get buffer limit is only called in the context of the set callbacks calls?

Reply:
I'm skeptical it will be called willy nilly but I'd be happy to have scope guards for setBufferLimits if you'd like.  I don't think we ought to for getting buffer limits - there could be a filter which did not latch limits which wanted to check to see if another filter changed it.

Reply:
I don't feel that strongly about it. Either way. Was just a random thought that might help find bugs later.